jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -405,6 +439,28 @@ object XGBoost extends Serializable {
     xgBoostModel.setPredictionCol(predCol)
   }
 
+  private[spark] def loadPrevBooster(sparkContext: SparkContext, hdfsTmpPath: String):
+      Booster = {
+    val fs = FileSystem.get(sparkContext.hadoopConfiguration)
+    if (hdfsTmpPath == null || !fs.exists(new Path(hdfsTmpPath))) {
+      null
+    } else {
+      val versions = fs.listStatus(new Path(hdfsTmpPath)).map(_.getPath.getName).collect {
+        case fileName if fileName.endsWith(modelSuffix) => fileName.stripSuffix(modelSuffix).toInt
+      }
```

Comment:
in some cases (though rare in XGBoost case), when we have a lot of files under the same directory, this listStatus can be very slow...

so we may want to clean those old files for each time we successfully load the latest checkpoint

Reply:
I think clean old files during saving makes more sense, because we actually can keep exactly 1 model for job recovering.